gap/callwithtimeout.gd

```diff
@@ -0,0 +1,54 @@
+#############################################################################
+##
+#F  IO_CallWithTimeout( <timeout>, <func>[, <arg1>[, <arg2>....]] )  
+##         . . call a function with a time limit
+#F  IO_CallWithTimeoutList( <timeout>, <func>, <arglist> )  
+##
+##  <#GAPDoc Label="IO_CallWithTimeout">
+##  <Index>Timeouts</Index>
+##  <ManSection>
+##  <Func Name="IO_CallWithTimeout" Arg='timeout, func, .....'/>
+##  <Func Name="IO_CallWithTimeoutList" Arg='timeout, func, arglist'/>
+##
+##  <Description>
+##    <C>IO_CallWithTimeout</C> and <C>IO_CallWithTimeoutList</C> support calling a function
+##  with a limit on length of time it will run. The function is run inside a copy of the current GAP
+##  session, so any changes it makes to global variables are thrown away when the function finishes
+##  or times out.<P/>
+##
+##  <C>IO_CallWithTimeout</C> is variadic. 
+##  Its third and subsequent arguments, if any, are the arguments passed to <A>func</A>.
+##  <C>IO_CallWithTimeoutList</C> in contrast takes exactly three arguments, of which the third is a list
+##  (possibly empty) or arguments to pass to <A>func</A>. <P/>
+##
+##  If the call completes within the allotted time and returns a value <C>res</C>, the result of 
+##  <C>IO_CallWithTimeout[List]</C> is a length 2 list of the form <C> [ true, res ] </C>. <P/>
+##  
+##  If the call completes within the allotted time and returns no value, the result of 
+##  <C>IO_CallWithTimeout[List]</C> is a list of length 1 containing the value <C>true</C>.<P/>
+##
+##  The call occurs with 
+##  If the call does not complete within the timeout, the result of <C>IO_CallWithTimeout[List]</C>
+##  is a list of length 1 containing the value <C>false</C>  In this case, just as if you had
+##  <C>quit</C> from a break loop, there is some risk that internal data structures in &GAP; may
+##  have been left in an inconsistent state, and you should proceed with caution.<P/>
+##
+##  The timer is suspended during execution of a break loop and abandoned when you quit from a break loop.<P/>
```

Comment:
Huh... so if the execution of the code happens in forked GAP, what happens with those break loops? Aren't they run in the forked GAP? If so, what happens if I "quit" the break loop -- do we exit from the forked process? Or are we stuck in it?

Reply:
(Just to put here), I still haven't fixed this. However, this exact same problem occurs in all the IO fork related functions (ParListByFork for example), so this isn't a new issue, and if it's going to get fixed it should be fixed more generally.

On the other hand, I'd quite like this function to be in IO, as it's (for me) the best current replacement for CallWithTimeout, which I often want.